File: .setup/create.untrusted.users.pl
#!/usr/bin/perl
use strict;
use warnings;
my $uuid=0;

foreach my $u (0..9) 
{
  $uuid=900+$u;
  system ("addgroup untrusted0$u --gid $uuid");
  system ("adduser untrusted0$u --home /tmp --no-create-home --uid $uuid --gid $uuid --disabled-password --gecos 'untrusted'");
}

foreach my $u (10..59)
{
  $uuid=900+$u;
  system ("addgroup untrusted$u --gid $uuid");
  system ("adduser untrusted$u --home /tmp --no-create-home --uid $uuid --gid $uuid --disabled-password --gecos 'untrusted'");
}

File: .setup/bin/create_untrusted_users.py
#!/usr/bin/env python3
"""
This creates untrusted users for the system starting at untrusted00 and going up to untrusted59.
We use these users to then run our autograding tests against student submissions in a tmp directory.
These users have very little permissions and access to the system at large.
"""
import os
import pwd


for i in range(0, 60):
    user = "untrusted" + str(i).zfill(2)
    try:
        pwd.getpwnam(user)
    except KeyError:
        uuid = 900 + i
        os.system("addgroup {} --gid {}".format(user, uuid))
        os.system("useradd --home /tmp -M --uid {} --gid {} "
                  "-c 'untrusted' {}".format(uuid, uuid, user))
